File: cash.c
#include <cs50.h>
#include <stdio.h>
int main(void)
{
    // ask for change owed in cents
    // repeat asking if it is less than 0
    int cents;
    do
    {
        cents = get_int("Change owed: ");
    }
    while (cents < 0);
    // declaring the variables to counts amount of each coin per type of coin
    int qnumber = 0;
    int dnumber = 0;
    int nnumber = 0;
    int pnumber = 0;
    // calculate amount of quarters needed
    while (cents >= 25)
    {
        qnumber++;
        cents -= 25;
    }

    // calculate the amount of dimes you can give
    while (cents >= 10)
    {
        dnumber++;
        cents -= 10;
    }

    // calculate the amount of nickels you can give
    while (cents >= 5)
    {
        nnumber++;
        cents -= 5;
    }

    // calculate the amount of pennies you should give
    while (cents >= 1)
    {
        pnumber++;
        cents -= 1;
    }
    // add up the numbers of each kind of coin and then print that number

    int coins = qnumber + dnumber + nnumber + pnumber;
    printf("%i\n", coins);
}

File: cash.py
from cs50 import get_float

dollars = get_float("Change: ")

cents = dollars * 100
coins = 0
#check quarters
while (cents >= 25):
    cents -= 25
    coins += 1

#check dines
while (cents >= 10):
    cents -= 10
    coins += 1

#check nickels
while (cents >= 5):
    cents -= 5
    coins += 1

#check pennies
while (cents >= 1):
    cents -= 1
    coins += 1

#print min coins
print(coins)
